gst/droidcodec/gstdroidvdec.c

```diff
@@ -1143,16 +1150,13 @@ gst_droidvdec_finish (GstVideoDecoder * decoder)
   GST_VIDEO_DECODER_STREAM_UNLOCK (decoder);
```

Comment:
Gstreamer locking is hard to follw at the best of times so I may be way off here, but I'm wondering if the problem isn't really that _frame_available is locking the stream mutex instead of something independent. I kind of hope I'm wrong because adding more mutexes isn't going to aid in understanding, but unlocking the stream mutex here, relocking it, and then carrying on as though no other state has changed seems wrong and I think the comments above a little about finish being able to be called again and returning an error in that case suggest there's already a problem with that and it's being worked around in an inadequate fashion.